src/tiled/mainwindow.cpp

```diff
@@ -816,6 +817,25 @@ bool MainWindow::saveFileAs()
     return saveFile(fileName);
 }
 
+void MainWindow::saveAll()
+{
+    foreach (MapDocument *mapDoc, mDocumentManager->documents()) {
+        QString fileName(mapDoc->fileName());
+        QString error;
+
+        if (fileName.isEmpty()) {
+            mDocumentManager->switchToDocument(mapDoc);
+            if (!saveFileAs())
+                return;
+        } else if (!mapDoc->save(fileName, &error)) {
+            QMessageBox::critical(this, tr("Error Saving Map"), error);
```

Comment:
I guess before showing this error, it also makes sense to switch to this document.